### ISSUE TITLE
Passkey not found

### ISSUE THREAD

**Issue by TheRavenNL** [bitwarden/clients#6764]
### Steps To Reproduce

1. Go to 'google.com'
2. Create and store a passkey in Bitwarden
3. Close the browser
4. Start the browser and go to 'google.com' and try to login
 with a passkey

### Expected Result

Passkey pops up in the plugin window

### Actual Result

Passkey is not found

### Additional Context

The passkey is visible in the vault.

### Operating System

Windows

### Operating System Version

Windows 11

### Web Browser

Microsoft Edge

### Browser Version

118.0.2088.76

### Build Version

2013.10.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Neonwarden** [comment]
Hi there,

I attempted to reproduce your issue and was unable to do so. Is your vault unlocked at the time of signing into Google? 

We use GitHub issues as a place to track bugs and other development related issues. If your issue persists, please write us back using our contact form, so we can continue troubleshooting: https://bitwarden.com/contact/ 

You can include a link to this issue in the message content.

Alternatively, you can also search for an answer in our help documentation (https://bitwarden.com/help/) or get help from other Bitwarden users on our community forums (https://community.bitwarden.com/c/support/). 

The issue here will be closed.

Thanks!

**Comment by kzzalews** [comment]
@Neonwarden I can reproduce the issue with google.com on my side. With the same configuration (Win11 and Edge). Would you like me to gather more info? How can I do that?

**Comment by thericle** [comment]
I have the exact same issue as described in the 2 posts above. I use Google Chrome on Windows 11.

**Comment by sammbw** [comment]
Hi there,

Thanks for your additional reports.

I was able to reproduce this issue myself, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Thanks once again!

**Comment by TheCrazyMax** [comment]
> Hi there,
> 
> Thanks for your additional reports.
> 
> I was able to reproduce this issue myself, and I have flagged this to our engineering team.
> 
> If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.
> 
> Thanks once again!

Thanks, I'm happy that my video recording sent by email helped to reproduce the issue :)

**Comment by OldMacDonald1** [comment]
I have noted the same problem.
It only affects MS Windows when loging in to my Google account.  I can login to Google using the passkey stored in BW successfully on MacOS and also Linux Debian, but fails when using Windows.

![image](https://github.com/bitwarden/clients/assets/128412667/a014b787-05f8-4b49-a77e-4e5f5c0620f3)

**Comment by ttallberg** [comment]
Same problem here logging in to google account on Windows 11 with Edge browser. Works OK on macOS Sonoma and Edge.

**Comment by jtackaberry** [comment]
Same behavior here with Windows 10 22H2 and Firefox 119 with Bitwarden 2023.10.1 installed (as a sideload via about:debugging since Mozilla hasn't approved 2023.10 on Mozilla Addons yet).

**Comment by kreake** [comment]
Same here (Win 11, Edge 119.0.2151.44, Bitwarden 2023.10.1):
![image](https://github.com/bitwarden/clients/assets/13981717/e0a28133-e15f-41b7-ba88-e2978ce32745)

**Comment by SergeyBashashin** [comment]
> ### Steps To Reproduce
> 1. Go to 'google.com'
> 2. Create and store a passkey in Bitwarden
> 3. Close the browser
> 4. Start the browser and go to 'google.com' and try to login
>    with a passkey
> 
> ### Expected Result
> Passkey pops up in the plugin window
> 
> ### Actual Result
> Passkey is not found
> 
> ### Screenshots or Videos
> _No response_
> 
> ### Additional Context
> The passkey is visible in the vault.
> 
> ### Operating System
> Windows
> 
> ### Operating System Version
> Windows 11
> 
> ### Web Browser
> Microsoft Edge
> 
> ### Browser Version
> 118.0.2088.76
> 
> ### Build Version
> 2013.10.0
> 
> ### Issue Tracking Info
> * [x]  I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

having exactly the same issue

**Comment by tomasztrzos** [comment]
I have exactly same issue on macos.

**Comment by isaumya** [comment]
I am also facing this exact same issue on Win 11 + Chrome.

Moreover, as the Bitwarden Android App update is here which now supports passkeys, I even tried with my Android Chrome Incognito window. Same Google issue remains on my phone as well.

**Comment by kriswilk** [comment]
> I am also facing this exact same issue on Win 11 + Chrome.
> 
> Moreover, as the Bitwarden Android App update is here which now supports passkeys, I even tried with my Android Chrome Incognito window. Same Google issue remains on my phone as well.

Correct me if I'm wrong, but I don't think the Android app can do anything with passkeys yet, other than show that you've saved one in a vault entry. Mobile passkey functionality is coming in a future update.

**Comment by isaumya** [comment]
> Correct me if I'm wrong, but I don't think the Android app can do anything with passkeys yet, other than show that you've saved one in a vault entry. Mobile passkey functionality is coming in a future update.

Sorry! That's my bad. The Android app shows the passkey for the login item but actually doesn't work. It just shows that passkey is there for that login. That's it.

**Comment by davxy** [comment]
I face this issue when using 2FA with security key (Yubikey).
After that I inserted user and pass. The pop-up appears.
The expected behavior was to touch the key  to unlock it

**Comment by ybce** [comment]
Same issue on MacOS, on sites where I used to use my fingerprint as a second factor, the popup comes up and the system fingerprint prompt doesn't show anymore. I didn't save any passkeys in the past in bitwarden related to these sites.

**Comment by kriswilk** [comment]
> Same issue on MacOS, on sites where I used to use my fingerprint as a second factor, the popup comes up and the system fingerprint prompt doesn't show anymore. I didn't save any passkeys in the past in bitwarden related to these sites.
> 
> Edit: Workaround seems to click `Use browser` in the bottom left corner of the popup, this brings up the system prompt for fingerprint and I assume for yubikey as well.

This is not the issue we're discussing here. This thread is about the situation where a user creates a passkey for Google, has it saved in BW, but BW reports "no passkey found" when subsequently trying to log into Google.

**Comment by clrizzi** [comment]
Same problem here, using Firefox and Bitwarden on Windows.

**Comment by TheUntouchable** [comment]
Okay, and i thought its just me..
Same on FF 119.0.1 and Bitwarden Extension 2023.10.1.
Else its working quite well with Github and other sites.

**Comment by johnnykang** [comment]
2023.10.2 chrome extension is out. Seems to 'partially' fix the Google passkey issue. I still can't get the account.google.com to detect the passkey on 'bitwarden' client, but it does fallback to Windows Hello or Yubikey now.

**Comment by TheUntouchable** [comment]
> [Browser v2023.10.2](https://github.com/bitwarden/clients/releases/tag/browser-v2023.10.2)
> 
>     Automatically fall back to the native "use browser" option when no passkeys exist in Bitwarden for a given website. The vault must be unlocked for us to know no passkeys exist.
>     Fall back to the native "use browser" option when closing the Bitwarden passkey window.
>     Respect excluded domains list (under Settings) before showing Bitwarden passkey options for a given website.
>     Fixes to iCloud and Google passkey login. There is still a known issue with Google on some Windows device configurations.
https://github.com/bitwarden/clients/releases

**Comment by kriswilk** [comment]
> 2023.10.2 chrome extension is out. Seems to 'partially' fix the Google passkey issue. I still can't get the account.google.com to detect the passkey on 'bitwarden' client, but it does fallback to Windows Hello or Yubikey now.

My browser extension also updated to 2023.10.2 (Edge, Windows 11). Now BW doesn't come up at all, only a QR code presented by the browser allowing me to sign in from the device that hosts the passkey.

This new behaviour is in line with what @kspearrin announced on reddit [here](https://www.reddit.com/r/Bitwarden/comments/17psqx6/bitwarden_launches_passkey_management/). Specifically:

"Automatically fall back to the native "use browser" option when no passkeys exist in Bitwarden for a given website. The vault must be unlocked for us to know no passkeys exist."

Basically, BW still thinks there's no passkey for google but now it's just not announcing it in a dialog box.

Waiting for a proper fix...

**Comment by isaumya** [comment]
Not sure what Google did which is so different that Passkey is only broken for Google. This is so weird. This update is a band-aid fix.

**Comment by TheCrazyMax** [comment]
> 2023.10.2 chrome extension is out. Seems to 'partially' fix the Google passkey issue. I still can't get the account.google.com to detect the passkey on 'bitwarden' client, but it does fallback to Windows Hello or Yubikey now.
> 
> PS:
> 
> When setting up the Passkey in the Account setting page, it is now able to add and detect and verify the Bitwarden passkey. However, when it is on the login page (account.google.com, where you input your username and choose passkey to proceed), Bitwarden passkey is NOT detected, and it falls back to the native implementation, Windows Hello or Yubikey. Just NOT Bitwarden extension.

That's true for the fallback. But the behaviour was already there on 2023.10.0 for the account setting page finding it, but not the login page :)

**Comment by TheCrazyMax** [comment]
> Not sure what Google did which is so different that Passkey is only broken for Google. This is so weird. This update is a band-aid fix.

The passkey from Google is not different form others. And other password managers or Windows Hello etc are handling it correctly. I don't know what the issue is exactly, but Google is sending the correct relying party ID and Bitwarden is not finding it on the login page for now.

"fido2Credentials":[
    {
        "credentialId":"***",
        "keyType":"public-key",
        "keyAlgorithm":"ECDSA",
        "keyCurve":"P-256",
        "keyValue":"***",
        "rpId":"google.com",
        "userHandle":"***",
        "counter":"3",
        "rpName":"Google",
        "userDisplayName":"***",
        "discoverable":"true",
        "creationDate":"2023-11-02T15:33:07.027Z"
    }
]

**Comment by OldMacDonald1** [comment]
> The passkey from Google is not different form others. And other password managers or Windows Hello etc are handling it correctly. I don't know what the issue is exactly, but Google is sending the correct relying party ID and Bitwarden is not finding it on the login page for now.
> 
> "fido2Credentials":[ { "credentialId":"[removed]", "keyType":"public-key", 

Can you repeat that on the 
      Manage Your Google Account->Security->Passkeys 
page because the "Give it a try" test button does work even on Windows.

**Comment by trmartin4** [comment]
We have been investigating this internally, and we believe that this behavior is due to an inconsistent implementation of WebAuthn on the part of Google.  We are still researching, but our current prerequisites for this behavior are:

1. The affected user is running Windows
2. The account has both a Security Key (non-discoverable, 2FA key) **and** a Passkey (discoverable) registered on their Google account, with the Security Key not in Bitwarden and the Passkey in Bitwarden.

In that scenario, Google does not properly supply the credential ID for the registered Passkey when you attempt to authenticate.  The only credential ID that is supplied is for the 2FA Security Key, **not** the Passkey.  This results in the Bitwarden extension showing "No credential found" (in `2023.10.1`) or reverting to the default browser behavior (in `2023.10.2`).  If no credential ID were supplied, Bitwarden would match on relying party ID, but per the WebAuthn spec if a credential is supplied the authenticator must respect that list and not provide credentials that are not on the list.

Interestingly, this behavior is not exhibited on the "Give it a try" button, so the credential is found in that case.

We are still actively researching this and will update if there are any updates.

**Comment by isaumya** [comment]
Thanks a lot for the explanation @trmartin4. I truly hope you and the team find a resolution soon. We all are eagerly waiting. I know it is insanely hard to work with Google as there is no way to contact them and ask them why they are doing this weird thing. They just do it and don't care about the rest.

**Comment by TheCrazyMax** [comment]
> > The passkey from Google is not different form others. And other password managers or Windows Hello etc are handling it correctly. I don't know what the issue is exactly, but Google is sending the correct relying party ID and Bitwarden is not finding it on the login page for now.
> > "fido2Credentials":[ { "credentialId":"[removed]", "keyType":"public-key",
> 
> Can you repeat that on the Manage Your Google Account->Security->Passkeys page because the "Give it a try" test button does work with BW (even on Windows). Then you can compare working and non working fido2credentials

Yes that's what I'm saying from the beginning :) And I told that to Bitwarden support to make them reopen this issue :)
But the relying party is "google.com" so it should work on both, but Bitwarden is not detecting it on the login page

**Comment by TheCrazyMax** [comment]
> We have been investigating this internally, and we believe that this behavior is due to an inconsistent implementation of WebAuthn on the part of Google. We are still researching, but our current prerequisites for this behavior are:
> 
> 1. The affected user is running Windows
> 2. The account has both a Security Key (non-discoverable, 2FA key) **and** a Passkey (discoverable) registered on their Google account, with the Security Key not in Bitwarden and the Passkey in Bitwarden.
> 
> In that scenario, Google does not properly supply the credential ID for the registered Passkey when you attempt to authenticate. The only credential ID that is supplied is for the 2FA Security Key, **not** the Passkey. This results in the Bitwarden extension showing "No credential found" (in `2023.10.1`) or reverting to the default browser behavior (in `2023.10.2`). If no credential ID were supplied, Bitwarden would match on relying party ID, but per the WebAuthn spec if a credential is supplied the authenticator must respect that list and not provide credentials that are not on the list.
> 
> Interestingly, this behavior is not exhibited on the "Give it a try" button, so the credential is found in that case.
> 
> We are still actively researching this and will update if there are any updates.

Same as @kriswilk here, no security key linked to my Google Account, only passkeys.

**Comment by limisk** [comment]
When I spoof my user agent to MacOS and clear my cookies (if using the extension "User-Agent Switcher and Manager" you also have to remove accounts.google.com from the blacklist), I can sign in using my passkey. So this confirms that Google changes their behavior based on the user's OS.

**Comment by trmartin4** [comment]
@TheCrazyMax @kriswilk, would you be able to share the number of passkeys you have registered on your Google accounts and stored outside of Bitwarden, and whether they are platform credentials or cross-platform credentials?  We have not been able to replicate that behavior but I would like to make sure we have exactly the same replication steps, as we are relying on the behavior of Google in their status as the relying party and we can't anticipate how their responses may differ and what criteria they may be using.

**Comment by limisk** [comment]
Even though I have not manually added any additional passkeys, there is an "automatically created passkey" on my Android phone that I cannot delete.

**Comment by limisk** [comment]
Okay, so after signing out on my phone, the automatically created passkey is gone, and now I can use my Bitwarden passkey! It might be that on Windows, the automatically created one takes precedence, and on MacOS, the custom one does, because Windows does not officially support passkeys yet.

**Comment by TheUntouchable** [comment]
All android phones, don't ask with which android version this started, are by default passkeys when they are linked to the google account as far as I know.

**Comment by limisk** [comment]
Strangely, the virtual authenticator from Chrome Devtools works with the same setup. When logging calls to navigator.credentials.get WITHOUT the virtual authenticator enabled, allowCredentials only contains one element, but when enabling the virtual authenticator, it contains two public key elements.

**Comment by limisk** [comment]
I think I found the culprit. It's the transport. When it's set to usb, ble, or nfc, it works, but with internal, it doesn't.

**Comment by TheCrazyMax** [comment]
> Okay, so after signing out on my phone, the automatically created passkey is gone, and now I can use my Bitwarden passkey!
> 
> I signed back in on my phone, and as expected, the automatically created passkey is back and I can't sign in on Windows anymore.
> 
> It might be that on Windows, the automatically created one takes precedence, and on MacOS, the custom one does. Because Windows does not officially support passkeys yet, Google might expect that users want to use cross device authentication with their Android phone there. (Just a theory.)

Windows is supporting only device bound ones, but they are supported since 23H2, they were there before and now we can manage them.
I don't think that this is a cause, as if you try it with 1Password, it works there without having to sign out from anywhere.

**Comment by TheCrazyMax** [comment]
> @TheCrazyMax @kriswilk, would you be able to share the number of passkeys you have registered on your Google accounts and stored outside of Bitwarden, and whether they are platform credentials or cross-platform credentials? We have not been able to replicate that behavior but I would like to make sure we have exactly the same replication steps. We are relying on the behavior of Google in their role as the relying party and we can't anticipate how their responses may differ and what criteria they may be using.

Sure, here is a screenshot
![image](https://github.com/bitwarden/clients/assets/15264289/11578f2c-eb19-48a9-8205-599cf7a2db32)
I have the automatically created one on Android, one on Windows, one on 1Password (as a trial to compare) and one for Bitwarden

**Comment by TheUntouchable** [comment]
For me its looking like this, also not working:

![grafik](https://github.com/bitwarden/clients/assets/2162540/87cd2101-dcc7-45af-9f27-e725f18afded)

**Comment by kriswilk** [comment]
> @TheCrazyMax @kriswilk, would you be able to share the number of passkeys you have registered on your Google accounts and stored outside of Bitwarden, and whether they are platform credentials or cross-platform credentials?

Coming back to the thread late but wanted to provide this info. My Google passkey page looks like this:

![Web capture_10-11-2023_75933_myaccount google com](https://github.com/bitwarden/clients/assets/382436/26727668-4253-4e31-98bf-b50d0175e72d)

I see that there's a potential "fix" in #6847. Looking forward to it.

I have to wonder how this problem got overlooked before launching passkey support. Windows/Google has got to be one of the most common combinations that anyone would use.

**Comment by limisk** [comment]
I also wonder what the benefit of doing this is for Google? I don't know how sending multiple allowed credentials could cause problems with a correct passkey implementation. Is it just a workaround for some browser/OS/Windows Hello bug on Google's part?

**Comment by TheCrazyMax** [comment]
> I also wonder what the benefit of doing this is for Google? I don't know how sending multiple allowed credentials could cause problems with a correct passkey implementation. Is it just a workaround for some browser/OS/Windows Hello bug on Google's part?

Here is the specification from W3C: https://w3c.github.io/webauthn/#dom-publickeycredentialrequestoptions-allowcredentials & https://w3c.github.io/webauthn/#dom-publickeycredentialdescriptor-transports
"If the [user account](https://w3c.github.io/webauthn/#user-account) to authenticate is already identified (e.g., if the user has entered a username), then the [Relying Party](https://w3c.github.io/webauthn/#relying-party) SHOULD use this member to list [credential descriptors for credential records](https://w3c.github.io/webauthn/#credential-descriptor-for-a-credential-record) in the [user account](https://w3c.github.io/webauthn/#user-account). This SHOULD usually include all [credential records](https://w3c.github.io/webauthn/#credential-record) in the [user account](https://w3c.github.io/webauthn/#user-account)."

=> It looks like it's due to Google asking the username, which is then using the same method as for non-discoverable credentials to find it, so it is highly recommended to specify the existing credentials in that scenario apparently

And yes, it looks like the "internal" value for transports may be an issue: https://w3c.github.io/webauthn/#dom-authenticatortransport-internal if Bitwarden is considered as a removable authenticator. Is it the case ?

**Comment by limisk** [comment]
Okay, no, it actually is a bug in Bitwarden, I think. This logic seems suspect:
https://github.com/bitwarden/clients/blob/ade9e9adfeb325bf5cbeac1b0bad0d7b3b344afa/libs/common/src/vault/services/fido2/fido2-authenticator.service.ts#L212

It should maybe use both findCredentialsById and findCredentialsByRp. With this, if there is a credential specified but it's not the right one, it does not try searching for credentialy by the relying party.

**Comment by limisk** [comment]
Oh, wait, this is bad: Fixing this would make Bitwarden non-compliant
https://w3c.github.io/webauthn/#dom-publickeycredentialrequestoptions-allowcredentials

**Comment by TheCrazyMax** [comment]
> Oh, wait, this is bad: "Fixing" this would violate the spec https://w3c.github.io/webauthn/#dom-publickeycredentialrequestoptions-allowcredentials "If not [empty](https://infra.spec.whatwg.org/#list-empty), the client MUST return an error if none of the listed credentials can be used."

To be checked, but I think that the Bitwarden fallback already implemented an error that redirects to the browser prompt. I don't remember in which commit I saw that.
But we may have to adapt it from "if no passkey found in Vault for RP" to "if no passkey from given list found in Vault"

**Comment by TheCrazyMax** [comment]
> > "if no passkey from given list found in Vault"
> 
> That is the way it is right now. And that is what causes it to fail with Google. Essentially, it would need to be "no passkey found in Vault or list" (which I think is equivalent to just ignoring allowCredentials entirely).
> 
> Right now it looks like one has to either change the transport to something incorrect (usb, ble, nfc) or just ignore allowCredentials. Both are workarounds that violate the spec.
> 
> The only remaining option would be to find another, hopefully compliant, way to get Google to send both credentials, or none at all. If Google sends just the Android one, to be compliant, Bitwarden has to reject the request.

Do we have any example of what Google is sending ? If they are following the standards, they should send all of them in the list sorted by preference. So a possible scenario is that Android is the first one in the list and that we are only checking for that one. It's just an assumption.

**Comment by limisk** [comment]
Yes, we do. I logged it with
`const old = navigator.credentials.get;
navigator.credentials.get = (x) => {
    console.log(x);
    debugger;
    return old.bind(navigator.credentials, x)();
};`
and when the conditions listed in my comment above are met, allowCredentials only has one member, and that is the "automatically created passkey".

**Comment by limisk** [comment]
The reason that Windows Hello works is that it just seems to ignore allowCredentials.

**Comment by TheCrazyMax** [comment]
> Windows Hello seems to do exactly what I proposed, completely ignoring allowCredentials. I tried intercepting it and only allowing a credential that I definitely do not have saved, and it still authenticates me. This does not seem to bother Google.
> 
> So, in essence, I think it might be worth diverging from the spec here. Windows Hello does so too, if I'm not wrong.

As Google is sending the username, that allows both a discoverable and a non-discoverable passkey to work there, so I'm wondering if Windows Hello is just using the userDisplayName ?
When a username is sent, would that be a solution here as well ?

**Comment by limisk** [comment]
Perhaps, but as I read the spec, even if there is a credential with a matching username, we are still not allowed to use it if it is not in allowCredentials.

**Comment by limisk** [comment]
A compromise could be allowing non listed credentials, but showing some UI indication that the website does not prefer you use this one. Or, if there are matching credentials in the vault, but none of them are in the list, only then ignore it.

**Comment by TheCrazyMax** [comment]
> Perhaps, but as I read the spec, even if there is a credential with a matching username, we are still not allowed to use it if it is not in allowCredentials.

It looks like the same behavior has already been observed and that apparently, from the answer, some standard may not be followed
Not sure what the best solution is

**Comment by TheCrazyMax** [comment]
@limisk From there https://www.w3.org/TR/webauthn-2/#sctn-terminology
I see that 
![image](https://github.com/bitwarden/clients/assets/15264289/99b5848f-2687-411d-b5e3-17e2239d8307)
So for me, it means that on Bitwarden we are storing discoverable credentials for Google, but Google is sending the request in a non-discoverable mode. 
But the spec is saying that it should be usable in both cases, so my guess is that we could safely ignore the allowCredentials for Google and look for a passkey that matches both the rpId "google.com" and the userDisplayName which is the account email address stored in the passkey.

If we match the rpId, it should be safe as it's the mechanism that is there to avoid phishing and other kind of attacks.
Can you easily try that solution on your side ?

**Comment by TheCrazyMax** [comment]
@limisk I'm also wondering about the "internal" value that you were speaking about: this demo site is the best that I found a few months ago and it looks like it does not work with Bitwarden because it can not read the "authenticatorAttachment" value, and Bitwarden is not going to the Fallback method there. 

=> Isn't it also a part of the issue, that we may not be registered as a correct authenticator attachment type ?
If I disable the Bitwarden extension, it's going to the Windows Hello Prompt and it works

**Comment by limisk** [comment]
I just found that the line saying "if there is no allowed credential the client must return an error" was only added in WebAuthn 3, which is still a draft. In WebAuthn 2 this is simply left unspecified, and I guess it could be interpreted either way. I don't think the note at [w3.org/TR/webauthn-2/#sctn-terminology](https://www.w3.org/TR/webauthn-2/#sctn-terminology) conflicts with either interpretation, as it could just be taken to mean "discoverable credentials can be used, if they are in the list".

> and look for a passkey that matches both the rpId "google.com" and the userDisplayName which is the account email address stored in the passkey.
The display name should never be used to look for a key. That's what the id field is for. But yes, that could be useful.

Either way, I'm not a Bitwarden developer, and I don't think there's any clearly correct option here, so I don't have much to add anymore.

**Comment by TheCrazyMax** [comment]
Ok thank you :)

@trmartin4 Can you help with that ?

**Comment by limisk** [comment]
> I'm also wondering about the "internal" value that you were speaking about: this demo site [webauthn.me](https://webauthn.me/) is the best that I found a few months ago and it looks like it does not work with Bitwarden because it can not read the "authenticatorAttachment" value, and Bitwarden is not going to the Fallback method there.

Hmm, this might be a separate bug with the fallback code. Bitwarden definitely sends an authenticatorAttachment ("cross-platform").

**Comment by limisk** [comment]
I honestly don't know how useful this discussion is, I think the Bitwarden team is investigating this separately internally and they have probably arrived at the same conclusions as me :) I was just curious and had time on my hands, so I decided to look into it.

**Comment by trmartin4** [comment]
Thank you all for the feedback.  Everything you have identified corresponds with the research we have done internally.  We would prefer to understand the intent of the behavior from Google in this case before adjusting the `transports`.  Ignoring the `allowCredentials` when provided would or misrepresenting the `transports` would be a last resort, as we would like to obey the letter and spirit of the WebAuthn spec as much as possible in our implementation.

As for the issue on webauthn.me, I believe that is the same issue discussed [here](https://github.com/bitwarden/clients/issues/6769), and we will be able to address that separately.

**Comment by clrizzi** [comment]
> > Thank you all for the feedback. Everything you have identified corresponds with the research we have done internally. We would prefer to understand the intent of the behavior from Google in this case before adjusting the `transports`. Ignoring the `allowCredentials` when provided would or misrepresenting the `transports` would be a last resort, as we would like to obey the letter and spirit of the WebAuthn spec as much as possible in our implementation.
> > As for the issue on webauthn.me, I believe that is the same issue discussed [here](https://github.com/bitwarden/clients/issues/6769), and we will be able to address that separately.
> 
> Any update on this?

Yeah... It's fixed by #6847

**Comment by ronidee** [comment]
Same problem under:
* Ubuntu 24.04, most recent firefox deb version,
* Android 14 (Lineage OS 21), Brave Nightly: "No sign-in info in Bitwarden"

Problem does **not** occur on my Ubuntu 23.10 installation using Firefox.

Here is a screenshot from the error message on Android:
![bitwardennosignininfo](https://github.com/bitwarden/clients/assets/22625272/aaf91e1d-2011-4d06-afe4-fa67613efd0b)

**Comment by ronidee** [comment]
Is this related to #6847? I.e. cisco does things the same as google causing Bitwarden to not find the passkeys? Sorry, I don't fully understand the technicalities of the issue this PR fixed, so I might be mistaken.

**Comment by ronhombre** [comment]
I encountered this issue again when trying to "Approve" a passkey in myaccount.google.com. It also doesn't work when trying to Approve using my phone with Bitwarden installed and fetching the passkey that way.

**Comment by pamperer562580892423** [comment]
@ronhombre Usually, I'm all in for "avoiding duplicate issues", but this issue here is almost three years old - and closed almost as long. Even if it's the same (or similar) symptoms - I would recommend to open a separate issue. Either a new bug report here on GitHub, [contacting BW](https://bitwarden.com/contact/) support or seeking advice in the [Community Forum](https://community.bitwarden.com) first.

**Comment by ronhombre** [comment]
The purpose of me replying here is because this is the direct way to
address the issue since it's extremely similar. I'm hoping for a discussion
with those involved, not to be shut down because of how long ago it was.

If you don't want the notification, just silence this issue.

**Comment by vinix00** [comment]
Same issue on my side. 
I'm on CachyOS with Helium Browser (Chromium).

**Comment by lkdm** [comment]
This does not seem to be fixed on Mac OS Firefox?
